### PR TITLE
👷‍♀️Test/api/cb/pwd/reset route

### PR DESCRIPTION
**Pull request by astroash**
- add test port in app
- add statusCodes for tested route
- create db queries for route testing
- add tape tests to check supertest expect errors

🤷‍♀️ not sure if this is best practise, I ended having to do a fair amount of db queries to be able to test routes

Also, I've not tested all the fail cases because the validation is painful and needs to be rewritten
relates #186

### CODE REVIEW

**Comment by astroash** [comment]
🙈

**Comment by astroash** [comment]
🙈

File: react-backend/app.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ app.use((err, req, res) => {
   res.send({ error, message });
 });
 
-const port = process.env.PORT || 5000;
+const port = process.env.NODE_ENV === 'test' ? 5010 : process.env.PORT || 5000;
```
Comment (eliascodes) [inline react-backend/app.js]: What is this for? So you can have the app running while you're running tests?

File: react-backend/routes/cbauthentication/cb_password_reset.js [hunk 2]
```diff
@@ -24,16 +24,16 @@ router.post('/', (req, res, next) => {
...
               pwdTokenAdd(token, tokenExpire, formEmail),
-              sendResetEmail(formEmail, token),
+              sendResetEmail(formEmail, token)
             ])
           )
           .then(() => res.send(exists))
           .catch(err => {
             console.log('Error sending email:', err);
-            res.send('failed');
+            res.status(501).send('failed');
```
Comment (eliascodes) [inline react-backend/routes/cbauthentication/cb_password_reset.js]: `501` is "not implemented". If you wanted a generic server error, that would be `500`. Although in this case, a `502` (bad gateway) is probably more appropriate.

File: tests/routes_tests/cbauthentication_tests/cb_password_reset.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    .end(async (err) => {
+      t.notOk(err, err || 'Passes supertest expect criteria');
+      try {
+        const getTokenFromDb = await dbConnection.query(
+          "SELECT * FROM cbusiness WHERE (token IS NOT NULL) AND email = 'findmyfroggy@frogfinders.com'",
+        );
+        t.ok(getTokenFromDb.rows.length, 'route successfully created a token in db');
+        t.end();
+      } catch (error) {
+        t.fail('Test query broke');
```
Comment (eliascodes) [inline tests/routes_tests/cbauthentication_tests/cb_password_reset.test.js]: You can just do `t.end(error)`, that way we get the stacktrace in the test output directly. This way we'd have to put in a `console.log` to find out what happened.

File: tests/routes_tests/cbauthentication_tests/cb_password_change.test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      t.notOk(err, err || 'Passes supertest expect criteria');
+      t.ok(res.body === true, 'matching passwords entered/token check on db returns true');
+      const oldPassword = '0a0429fa911712f7aca189bb12995963e3fc8f361e2845f747994be499250762';
+
+      try {
+        const newPassword = await dbConnection.query('SELECT hash_pwd FROM cbusiness WHERE id = 3');
+        t.notEqual(newPassword, oldPassword, 'Password has been changed in db');
+        t.end();
+      } catch (error) {
+        t.fail('Select test query broke');
```
Comment (eliascodes) [inline tests/routes_tests/cbauthentication_tests/cb_password_change.test.js]: See other comment on `t.end(error)`

File: tests/routes_tests/cbauthentication_tests/cb_password_change.test.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+test('POST /api/cb/pwd/change | token check/pw update successful', async (t) => {
+  await rebuild();
+  const token = await tokenGen();
+  const tokenExpire = Date.now() + 36000;
+
+  try {
+    const queryText = 'UPDATE cbusiness SET token = $1, tokenexpire = $2 WHERE id = 3';
+    await dbConnection.query(queryText, [token, tokenExpire]);
+  } catch (error) {
+    t.fail('Update test query broke');
```
Comment (eliascodes) [inline tests/routes_tests/cbauthentication_tests/cb_password_change.test.js]: See other comment on `t.end(error)`

File: react-backend/app.js (hunk 1 above)
Comment (astroash) [reply react-backend/app.js]: Yup

**Comment by astroash** [comment]
Ummm... dunno what happened there 🙈

File: react-backend/app.js (hunk 1 above)
Comment (RogeredBacon) [reply react-backend/app.js]: I stole this for my pr 😈

**Comment by eliascodes** [comment]
@astroash you rebased/force-pushed? Nice.

File: react-backend/routes/cbauthentication/cb_password_reset.js [hunk 6]
```diff
@@ -24,16 +24,16 @@ router.post('/', (req, res, next) => {
...
               pwdTokenAdd(token, tokenExpire, formEmail),
-              sendResetEmail(formEmail, token),
+              sendResetEmail(formEmail, token)
             ])
           )
           .then(() => res.send(exists))
           .catch(err => {
             console.log('Error sending email:', err);
-            res.send('failed');
+            res.status(418).send('failed');
```
Comment (eliascodes) [inline react-backend/routes/cbauthentication/cb_password_reset.js]: Come on 🍵 

`502` please

Comment (astroash) [reply react-backend/routes/cbauthentication/cb_password_reset.js]: 👼

**Comment by astroash** [comment]
> you rebased/force-pushed? Nice.
Maybe...

**Pull request merged**